src/hl/plist.rs

```diff
@@ -202,6 +203,20 @@ impl PropertyList {
     }
 }
 
+pub fn set_vlen_manager(plist: crate::internal_prelude::hid_t) -> Result<()> {
```

Comment:
Maybe add `// TODO: move this to dataset_transfer module when DatasetTransfer plist is implemented`? (so it doesn't hang out here in the wild)

Also, I don't think this name is clear enough; two alternative options: 

1. `pub fn set_vlen_manager_libc(plist: hid_t)`
2. `pub fn set_vlen_manager<A: Fn(usize) -> *mut u8, F: Fn(*mut u8)>(plist: hid_t, alloc: A, free: F)`

In (2), you'd call it as

```rust
set_vlen_manager(plist_id, |size| unsafe { libc::malloc(size) }, |ptr| unsafe { libc::free(ptr) })
```

(assuming the catch_unwind stuff can remain in the generic function itself)

You could probably even have both, where (1) calls (2).

Reply:
(2) requires quite a bit of trickery, as we have to get a function pointer from a closure. The easiest way would be to get a closure pointer, wrap this up in the alloc_info and free_info, and call this from a trampoline. This is a bit overkill when we are tied to the same allocator as for the varlen types in h5-types .

Reply:
Re: (2), yea, that's kind of what I was thinking, that's why the info is there :) (There's a similar example in Rustonomicon, I think)

The reason I mentioned it - we'll get to adding proper dxpl plists soon (there's just a few functions) - there we'll need to implement this in the most generic fashion.

For now it's ok I guess as long as there's a todo note.

Reply:
I have also done something similar in a c-wrapper for [`uvc`](https://github.com/mulimoen/libuvc-rs/blob/3394f0b7aecd315a23e0761f1cbd4ce713a413ff/src/streaming.rs#L51). It's a bit tricky to keep the lifetimes in order!